pyV2DL3/eventdisplay/util.py

```diff
@@ -71,7 +71,7 @@ def getGTI(BitArray, run_start_from_reference):
 
     n = BitArray.size
     TimeArray_s = [np.binary_repr(BitArray[i]).count("1") for i in range(n)]
-    duration_s = (n - 1) * 8 + TimeArray_s[-1]
+    duration_s = (n - 2) * 8 + TimeArray_s[-2] + TimeArray_s[-1]
```

Comment:
I am not sure I understand the solution. Might there be a case where you need to look into TimeArray_s[-3]?

Why are there two not completely filled bins in the end of the TimeArray_s? 

What happens when len(TimeArray) is smaller than 2?

Reply:
> I am not sure I understand the solution. Might there be a case where you need to look into TimeArray_s[-3]?
> 
No, that case does not happen! since this is for getting exact duration of run, and to make sure we are not adding extra seconds for the runs of this type (runs with only one time mask at the end with width <16 sec). The exact duration we need for filling the **stop time of the last GTI.** 

It is only for the run with such case: 
1. the duration falls in the **second last element** of TimeArray_s       
2. since (a.) there is only one time mask and (b.) for loop runs over entire array except first and last entry of  TimeArray_s, the `gti_end` array remain empty -> which should be duration_s




> Why are there two not completely filled bins in the end of the TimeArray_s?
> 
For this run time cut is 12 sec starting at 1170, and run duration is 1182 sec. So,`TBits` stored in anasum `maskBits` are 1182 bits with the first 1170 bits `1` and the last 12 bits as `0`, and we read  `maskBits` with uproot `member("fAllBits")`, which reads it as uint8. Now, the second last bin which is 3 (0000 0011), gets 6 `0` bits, but since the `TBits` ends at 1182,  the remaining 6 `0` bits are read with addition 2 '0' bits for reading uint8.  


> What happens when len(TimeArray) is smaller than 2?
 len(TimeArray) < 2 means <= 8 sec run 😰 ! We might need special care if we have such runs in our database and they need to be converted in DL3 !

Reply:
OK - forget about the 8s runs....

Wouldn't it be easier simply to remove all empty / zero elements at the end of the array? I still didn't complete understand the logic, but does it work when we have e.g., a 20 sec cut at the end of a run? This really should work for any foreseeable cases.

Reply:
Yes! that would be easier just to remove zero elements at the end of  TimeArray_s array.

Yes, it does work for time cut >= 16 sec! For eg. same run 74361, if there is cut like `1162 20 0`, will give  TimeArray_s as:

> array([255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255,
>        255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255,
>        255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255,
>        255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255,
>        255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255,
>        255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255,
>        255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255,
>        255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255,
>        255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255,
>        255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255,
>        255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255,
>        255, 255,   3,   0,   0], dtype=uint8)

and we get GTIs start and stop in second since run start: [0] [1162]

Reply:
This is actually the perfect function to write a unit test for...you could make sure this way that all of those issues are captured.

Reply:
- [ ] write unit test for getGTI()